src/main/java/net/dries007/tfc/common/recipes/HeatingRecipe.java

```diff
@@ -0,0 +1,130 @@
+package net.dries007.tfc.common.recipes;
+
+import java.util.Arrays;
+import java.util.Collection;
+import java.util.stream.Collectors;
+import javax.annotation.Nullable;
+
+import com.google.gson.JsonObject;
+import net.minecraft.item.Item;
+import net.minecraft.item.ItemStack;
+import net.minecraft.item.crafting.IRecipeSerializer;
+import net.minecraft.item.crafting.IRecipeType;
+import net.minecraft.item.crafting.Ingredient;
+import net.minecraft.item.crafting.ShapedRecipe;
+import net.minecraft.network.PacketBuffer;
+import net.minecraft.util.JSONUtils;
+import net.minecraft.util.ResourceLocation;
+import net.minecraft.world.World;
+
+import net.dries007.tfc.common.capabilities.heat.HeatCapability;
+import net.dries007.tfc.common.capabilities.heat.IHeat;
+import net.dries007.tfc.util.collections.IndirectHashCollection;
+
+public class HeatingRecipe extends SimpleItemRecipe
```

Comment:
Heat recipes can produce either items or fluids, this chosen to be unimplemented or forgotten?